build-system/common/git.js

```diff
@@ -27,32 +27,32 @@ const {
 } = require('./travis');
 const {getStdout} = require('./exec');
 
+/**
+ * Returns the remote tracking branch associated with the local master branch.
+ * @return {string}
+ */
+function gitUpstreamMaster() {
+  const upstreamName = getStdout('git config branch.master.remote').trim();
+  return (upstreamName || 'origin') + '/master';
+}
+
 /**
  * Returns the commit at which the current branch was forked off of master.
  * On Travis, there is an additional merge commit, so we must pick the first of
  * the boundary commits (prefixed with a -) returned by git rev-list.
- * On local branches, this is merge base of the current branch off of master.
+ * On local branches, this is merge base of the current branch off of
+ * upstream/master.
  * @return {string}
  */
 function gitBranchCreationPoint() {
   if (isTravisBuild()) {
     const traviPrSha = travisPullRequestSha();
+    const upstreamMaster = gitUpstreamMaster();
     return getStdout(
-      `git rev-list --boundary ${traviPrSha}...master | grep "^-" | head -n 1 | cut -c2-`
+      `git rev-list --boundary ${traviPrSha}...${upstreamMaster} | grep "^-" | head -n 1 | cut -c2-`
```

Comment:
This change from `master` to `${upstreamMaster}` was incorrect. This block has been removed in a subsequent commit, though.

Reply:
I don't think it's safe to replace `git rev-list` with `git merge-base`. See See #22355 for more context.

Reply:
Thanks I was able to reproduce the issue with multiple merges. PR significantly toned down from version you originally reviewed.

Reply:
Marking this comment as resolved since updated PR does not touch Travis code paths anymore.